crates/forge/src/partition.rs

```diff
@@ -0,0 +1,111 @@
+use serde::Serialize;
+use std::{collections::HashMap, str::FromStr};
+
+#[derive(Debug, Clone, Copy, Serialize)]
+pub struct Partition {
+    index: usize,
+    total: usize,
+}
+
+#[allow(dead_code)] // TODO: Removed in later PRs
```

Comment:
Doesn't really matter here but except is safer than allow because it will fail if the condition (dead_code in this case) is not met

```suggestion
#[except(dead_code)] // TODO: Removed in later PRs
```